File: Assignment_2(Android)/app/src/main/java/au/edu/utas/kit305/tutorial05/PlayerStats.kt
package au.edu.utas.kit305.tutorial05

data class PlayerStats(
    val playerName: String,
    var totalKickGoals: Int, // Total kicks that scored goals (6 points)
    var totalKickBehinds: Int, // Total kicks that scored behinds (1 point)
    var totalKickNoScore: Int,
    var totalHandballBehinds: Int, // Total handballs that scored behinds (1 point)
    var totalMarks: Int,
    var totalTackles: Int,
    var totalScore: Int // Total score calculated based on actions
) {
    // Helper function to calculate the total score
    fun calculateTotalScore(): Int {
        return (totalKickGoals * 6) + totalKickBehinds + totalHandballBehinds
    }
}

File: assignment-4-HaoyuWang-star-main(Flutter)/kit305-Assignment4/kit305-Assignment4/lib/PlayerStats.dart
class PlayerStats {
  final String playerName;
  int totalKickGoals;
  int totalKickBehinds;
  int totalKickNoScore;
  int totalMarks;
  int totalTackles;
  int totalHandballBehinds;
  int totalHandballNoScore;

  PlayerStats({
    required this.playerName,
    this.totalKickGoals = 0,
    this.totalKickBehinds = 0,
    this.totalKickNoScore = 0,
    this.totalMarks = 0,
    this.totalTackles = 0,
    this.totalHandballBehinds = 0,
    this.totalHandballNoScore = 0,
  });

  int get totalScore {
    return (totalKickGoals * 6) + totalKickBehinds + totalHandballBehinds;
  }

  int get totalKicks => totalKickGoals + totalKickBehinds + totalKickNoScore;
  int get totalHandballs => totalHandballBehinds + totalHandballNoScore;
  int get totalBehinds => totalKickBehinds + totalHandballBehinds;
}
